### PR TITLE
fedora-rhcos: Initial location to help kickstart further work

### PR DESCRIPTION
**Pull request by ashcrow**
Meant to help kick start RHCOS using Fedora content POC.

Items that can easily be reused are soft-linked in to the new sub-directory.

### CODE REVIEW

File: fedora-rhcos/overlay.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#       patches: drop
+#       branch: master
+
+  # Required as a cri-o dependency
+  - src: https://github.com/containernetworking/plugins.git
+    distgit:
+        name: containernetworking-cni
+        patches: drop
+
+  - src: github:openshift/redhat-release-coreos
```
Comment (cgwalters) [inline fedora-rhcos/overlay.yml]: I think we'll need to fork this one.

File: fedora-rhcos/rhcos.tdl [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<template>
```
Comment (cgwalters) [inline fedora-rhcos/rhcos.tdl]: This one can likely be a symlink too.

File: fedora-rhcos/host.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  - root
+ignore-removed-groups:
+  - root
+check-passwd:
+  type: "file"
+  filename: "passwd"
+check-groups:
+  type: "file"
+  filename: "group"
+packages:
```
Comment (cgwalters) [inline fedora-rhcos/host.yaml]: Let's add a big comment here like:
```
# The canonical version of this is in host.yaml - edit that first!
```

**Comment by ashcrow** [comment]
Updating shortly ... thanks!

File: fedora-rhcos/host.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+documentation: false
+# Doesn't quite work out of the box yet in RHEL7,
+# but see https://pagure.io/fedora-atomic/pull-request/62/edit
+tmp-is-dir: false
+initramfs-args:
+  - "--no-hostonly"
+  - "--add"
+  - "iscsi"
+automatic_version_prefix: "3.10-7.5"
+mutate-os-release: "7"
```
Comment (jlebon) [inline fedora-rhcos/host.yaml]: I guess these should be something like...

```yaml
automatic_version_prefix: "3.10-28"
mutate-os-release: "28"
```

?

**Comment by jlebon** [comment]
> I'm going back and forth on whether this should be a new repo or not...leaning a bit towards the latter. If we do that, one idea is to have rhcos inherit it as a git submodule or something? The main idea would be that if FCOS adds a package we should inherit that by default say.

I might be misunderstanding here. This PR seems to be about making RHCOS based on Fedora content, right? Or is this actually to start FCOS? I reviewed it under the assumption it was the former. If the latter, I definitely agree we should make it a separate repo!

Once we have FCOS going though, I do like the idea of having RHCOS just doing `include: fcos/host.yaml`, where `fcos/` is a submodule.

File: fedora-rhcos/host.yaml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  type: "file"
+  filename: "group"
+packages:
+ # Kernel + systemd
+ - kernel irqbalance microcode_ctl systemd
+ # Release package
+ - redhat-release-coreos
+ # rpm-ostree
+ - rpm-ostree nss-altfiles
+ # Subscriptions
```
Comment (miabbott) [inline fedora-rhcos/host.yaml]: Probably could drop these `subscription-manager` packages for FCOS

File: fedora-rhcos/host.yaml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  type: "file"
+  filename: "passwd"
+check-groups:
+  type: "file"
+  filename: "group"
+packages:
+ # Kernel + systemd
+ - kernel irqbalance microcode_ctl systemd
+ # Release package
+ - redhat-release-coreos
```
Comment (miabbott) [inline fedora-rhcos/host.yaml]: We'll likely need a `fedora-release-coreos`

File: fedora-rhcos/host.yaml [hunk 7]
```diff
@@ -0,0 +1,95 @@
+ref: "openshift/3.10/x86_64/os"
```
Comment (miabbott) [inline fedora-rhcos/host.yaml]: Do we want to differentiate the `ref` between RHCOS and FCOS?

**Comment by ashcrow** [comment]
> I might be misunderstanding here. This PR seems to be about making RHCOS based on Fedora content, right?

Correct.

> Or is this actually to start FCOS? I reviewed it under the assumption it was the former. If the latter, I definitely agree we should make it a separate repo!

It's not to start FCOS, though I'm sure there is overlap.

> Once we have FCOS going though, I do like the idea of having RHCOS just doing include: fcos/host.yaml, where fcos/ is a submodule.

That makes sense to me as well. However, at the time of this writing, we've been asked to have a Fedora version of RHCOS as part of this milestone.

File: fedora-rhcos/host.yaml (hunk 7 above)
Comment (ashcrow) [reply fedora-rhcos/host.yaml]: Keep in mind this is still RHCOS.

File: fedora-rhcos/host.yaml (hunk 3 above)
Comment (ashcrow) [reply fedora-rhcos/host.yaml]: Added

File: fedora-rhcos/rhcos.tdl (hunk 2 above)
Comment (ashcrow) [reply fedora-rhcos/rhcos.tdl]: Done

File: fedora-rhcos/host.yaml (hunk 7 above)
Comment (miabbott) [reply fedora-rhcos/host.yaml]: Oops...that invalidates most of my comments  ¯\_(ツ)_/¯

**Comment by cgwalters** [comment]
We're going to need naming conventions...I propose `rhcos-el7` for the current stream, and `rhcos-f28` for what's being discussed here.  I think this is better than `fedora-rhcos` because the "rhcos" aspect basically has higher "priority".  

One thing that's related to this is I think we should consider *requiring* the pivot...i.e. we only ship BootstrapOS.  If we go that approach, then a natural thing to do is to start with having `rhcos-f28` just be BootstrapOS for now.  (Shorthand: `rhbootcoreos-f28`?)

**Comment by cgwalters** [comment]
I'd like to avoid duplicating the package lists; we should be able to do that using the treefile `include:` feature.

**Comment by ashcrow** [comment]
To follow up a little here, Dusty and I spoke with Ben and we are pushing this goal out in favor of other goals. We still want to do this but having two pipelines for RHCOS isn't an activy priority right now.

**Pull request closed without merging**